packages/nextjs/app/blockexplorer/transaction/[txHash]/page.tsx

```diff
@@ -1,153 +1,19 @@
-"use client";
-
-import { useEffect, useState } from "react";
-import { useRouter } from "next/navigation";
+import TransactionComp from "../_components/TransactionComp";
 import type { NextPage } from "next";
-import { Hash, Transaction, TransactionReceipt, formatEther, formatUnits } from "viem";
-import { hardhat } from "viem/chains";
-import { usePublicClient } from "wagmi";
-import { Address } from "~~/components/scaffold-eth";
-import { useTargetNetwork } from "~~/hooks/scaffold-eth/useTargetNetwork";
-import { decodeTransactionData, getFunctionDetails } from "~~/utils/scaffold-eth";
-import { replacer } from "~~/utils/scaffold-eth/common";
+import { Hash } from "viem";
 
 type PageProps = {
   params: { txHash?: Hash };
 };
+
+export function generateStaticParams() {
+  return [{ txHash: "0x0000000000000000000000000000000000000000" }];
+}
 const TransactionPage: NextPage<PageProps> = ({ params }: PageProps) => {
-  const client = usePublicClient({ chainId: hardhat.id });
   const txHash = params?.txHash as Hash;
-  const router = useRouter();
-  const [transaction, setTransaction] = useState<Transaction>();
-  const [receipt, setReceipt] = useState<TransactionReceipt>();
-  const [functionCalled, setFunctionCalled] = useState<string>();
-
-  const { targetNetwork } = useTargetNetwork();
-
-  useEffect(() => {
-    if (txHash && client) {
-      const fetchTransaction = async () => {
-        const tx = await client.getTransaction({ hash: txHash });
-        const receipt = await client.getTransactionReceipt({ hash: txHash });
-
-        const transactionWithDecodedData = decodeTransactionData(tx);
-        setTransaction(transactionWithDecodedData);
-        setReceipt(receipt);
-
-        const functionCalled = transactionWithDecodedData.input.substring(0, 10);
-        setFunctionCalled(functionCalled);
-      };
-
-      fetchTransaction();
-    }
-  }, [client, txHash]);
 
-  return (
```

Comment:
Had to move all this in seprate component because `generateStaticParams` and "use client" cannot be present in same file